app/javascript/components/provider-form/index.jsx

```diff
@@ -12,23 +12,36 @@ import ProtocolSelector from './protocol-selector';
 import ProviderCredentials from './provider-credentials';
 import ValidateProviderCredentials from './validate-provider-credentials';
 import DetectButton from './detect-button';
+import debouncePromise from '../../helpers/promise-debounce';
 
 const findSkipSubmits = (schema, items) => {
   const found = schema.skipSubmit && items.includes(schema.name) ? [schema.name] : [];
   const children = Array.isArray(schema.fields) ? schema.fields.flatMap((field) => findSkipSubmits(field, items)) : [];
   return [...found, ...children];
 };
 
+const asyncValidator = (value = '', itemId) =>
+  API.get(`/api/providers?filter[]=name=${value}&expand=resources`)
+    .then((json) => {
+      if (json.resources.find(({ id, name }) => name === value && id !== itemId)) {
+        throw __('Name has already been taken');
+      }
+      if (value === '' || value === undefined) {
+        throw __('Required');
+      }
+      return undefined;
+    });
+
+const asyncValidatorDebounced = debouncePromise(asyncValidator);
+
 const commonFields = [
   {
     component: componentTypes.TEXT_FIELD,
     id: 'name',
     name: 'name',
     label: __('Name'),
     isRequired: true,
-    validate: [{
-      type: validatorTypes.REQUIRED,
-    }],
+    validate: [(value) => asyncValidatorDebounced(value, type)],
```

Comment:
the `type` is not received in here..

could you please change the function to `const commonFields = (type) => [` and call this from `loadProviderFields` like   `  ...commonFields(type),`